test/integration/rest-basic/index.spec.ts

```diff
@@ -64,12 +64,12 @@ test('rest editor basics', async ({ page, context, localApp }) => {
 
   await expect(newQueryEditor).toBeVisible();
 
-  await expect(editorModel.appCanvas.getByText('query4 authorization: foo')).toBeVisible();
+  await expect(editorModel.appCanvas.getByText('query4 authorization: foo')).toBeDefined();
```

Comment:
A locator is always defined. This will always pass. When writing tests, however trivial the change, best to always verify it still breaks when you expect it to.

```tsx
// this passes as well
await expect(editorModel.appCanvas.getByText("I'm not on the page at all")).toBeDefined();
```

Reply:
Yeah, I has thought of this and was gonna give it a second look - it's just that the dialog is covering the UI so I think this was making the test fail, I'll find the right way to do this.

Reply:
You could try `.toBeAttached`. I didn't think an obscured element would be omitted by `.toBeVisible`. I thought that was what `. toBeInViewport` did.

Any reason why this would only fail on windows?

Reply:
True that it's weird if it failed on Windows only...
I'll recheck as this might not be necessary, the test was being flaky locally sometimes so it might have been misleading.

Reply:
For this test, try with the instructions above and run httpbin in a container locally. That should help with the flakeyness. I'm planning to write a mock server into the test itself in the future.